text/0000-try-expr.md

```diff
@@ -0,0 +1,886 @@
+
+In the RFC author's opinion however, the sum total benefits of `try { .. }`
+seem to outweigh the drawbacks of the difficulty with purging [`try!`] from
+our collective memory.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Review considerations
+
+Among the considerations when picking a keyword are, ordered by importance:
+
+1. Fidelity to the construct's actual behavior.
+
+2. Precedent from existing languages
+    1. Popularity of the languages.
+    2. Fidelity to behavior in those languages.
+    3. Familiarity with respect to their analogous constructs.
+
+   See the [prior art][prior-art] the [rationale for try]
+   for more discussion on precedent.
+
+2. Brevity.
+
+[`Try`]: https://doc.rust-lang.org/nightly/std/ops/trait.Try.html
+
+4. Consistency with the naming of the trait used for `?` (the [`Try`] trait).
+
+5. Consistency with related standard library function conventions.
+
+6. Degree / Risk of breakage.
+
+7. Consistency with old learning material.
+
+    1. Inversely: The extent of the old learning material
+
+   That is, (in)consistency with `?` and the `try!()` macro.
+   If the first clause is called `try`,
+   then `try { }` and `try!()` would have essentially inverse meanings.
+
+## Rationale for `try`
+[rationale for try]: #rationale-for-try
+
+1. **Fidelity to the construct's actual behavior:** Very high
+2. **Precedent from existing languages:** A lot, see [prior-art]
+    1. **Popularity of the languages:** Massive accumulated dominance
+    2. **Fidelity to behavior in those languages:** Very high
+    3. **Familiarity with respect to their analogous constructs:** Very high
+3. **Brevity / Length:** 3
+4. **Consistency with the naming of the trait used for `?`:** Consistent
+5. **Consistency with related libstd fn conventions:** Consistent
+6. **Risk of breakage:** High (if we assume `try!` will break, otherwise: Low)
+    - **Used in std:** [*No*](https://doc.rust-lang.org/nightly/std/?search=try), `std::try!`, but it is technically possible to not break this macro. (unstable: `std::intrinsics::try` so irrelevant)
+    - **Used as crate?** [*Yes*](https://crates.io/crates/try). No reverse dependencies. Described as: *"Deprecation warning resistant try macro"*
+    - **Usage (sourcegraph):** **27** regex:
+    ```
+    repogroup:crates case:yes max:400
+    \b((let|const|type|)\s+try\s+=|(fn|impl|mod|struct|enum|union|trait)\s+try)\b
+    ```
+7. **Consistency with old learning material:** Inconsistent ([`try!`])
+
+### Review
+
+This is our choice of keyword, because it:
+
+1. has a massive dominance in both popular and less known languages and is
+   sufficiently semantically faithful to what `try` means in those languages.
+2. is consistent with the standard library wrt. `Try` and `try_` prefixed methods.
+3. it is brief.
+4. it has high fidelity wrt. the concepts it attempts to communicate (exception boundary for `?`).
```

Comment:
This is also true for `catch`?

Reply:
Somewhat yes I guess; I'll discuss this in a more nuanced fashion.